subprojects/base-services/src/main/java/org/gradle/internal/operations/DefaultBuildOperationQueue.java

```diff
@@ -129,20 +148,52 @@ private static String getFailureMessage(Collection<? extends Throwable> failures
         return "Multiple build operations failed.";
     }
 
+    private class SubmittedWorker {
```

Comment:
Ahhh, right.  Yes, this is much simpler.  Thanks for clarifying.  I've made those changes.  +100 to anything that keeps this stuff as simple as possible.